### PR TITLE
Add a fix for overflowing labels in owner picker in catalog + hover feature

### PR DESCRIPTION
**Pull request by JGoggers**
#### Old behaviour

Screenshot 2024-01-03 at 09 18 57

Note: The label is split for '-' but in case of consecutive letters, the label will not be split into different lines.
#

Two minor changes have been made.

#### Overflowing labels in OwnerPicker on the Catalog page are truncated.

Screenshot 2024-01-02 at 13 48 52</img>Screenshot 2024-01-02 at 13 48 52</img>Screenshot 2024-01-02 at 13 48 52</img>

#### Hovering over the labels in the OwnerPicker reveal the full label name. 

Screenshot 2024-01-02 at 13 49 03</img>

(Ideally all filter boxes would have this functionality to avoid overflow)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by alde** [comment]
looks good - do you have a screenshot of current behaviour for comparison?

File: plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx [hunk 1]
```diff
@@ -173,18 +177,25 @@ export const EntityOwnerPicker = (props?: EntityOwnerPickerProps) => {
...
+                        <GroupIcon fontSize="small" />
+                      ) : (
+                        <PersonIcon fontSize="small" />
+                      )}
+                      &nbsp;
+                      <Box
+                        style={{
+                          overflow: 'hidden',
+                          textOverflow: 'ellipsis',
+                          width: '11rem',
```
Comment (alde) [inline plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx:192]: `width: 11rem` seems like a bit of a magic number

File: plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx [hunk 2]
```diff
@@ -173,18 +177,25 @@ export const EntityOwnerPicker = (props?: EntityOwnerPickerProps) => {
...
-                    ) : (
-                      <PersonIcon fontSize="small" />
-                    )}
-                    &nbsp;
-                    {humanizeEntity(
-                      entity,
-                      humanizeEntityRef(entity, { defaultKind: entity.kind }),
-                    )}
-                  </Box>
+                  <Tooltip title={humanizedName}>
```
Comment (alde) [inline plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx:180]: could be good to look into using the presentation API instead

https://backstage.io/docs/reference/plugin-catalog-react.useentitypresentation/

File: plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx (hunk 1 above)
Comment (JGoggers) [reply plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx:192]: It honestly is. I'll try to change it to be more dynamic

File: plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx (hunk 2 above)
Comment (JGoggers) [reply plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx:180]: Will do! (I also added the screenshot for the previous behaviour) 

Thanks for the comments

Comment (JGoggers) [reply plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx]: I run into issues with hooks 
`React Hook "useEntityPresentation" cannot be called inside a callback`. 

If I disable eslint for this line, I get 
`Rendered more hooks than during the previous render`. 

Seems best to keep it as it is for this one

Comment (freben) [reply plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx]: Nah that's just how hooks work in React; you can for example make a wrapper component to put the hook in, or similar

Comment (JGoggers) [reply plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx]: Thanks for the tip, I changed it to use presentation API now

**Review by freben — APPROVED** [review]
Approving to unblock; catalog maintainers for final approve and merge

**Pull request merged**